src/libtls/tls_crypto.c

```diff
@@ -2426,3 +2426,51 @@ tls_named_group_t tls_ec_group_to_curve(diffie_hellman_group_t group)
 	}
 	return 0;
 }
+
+/**
+ * See header.
+ */
+array_t *tls_create_common_key_type(chunk_t hashsig, tls_version_t min_version,
+									tls_version_t max_version)
+{
+	bio_reader_t *reader;
+	array_t *types;
```

Comment:
Maybe an array is a bit of an overkill here... I use it to mimic a Set ADT that only contains each available key type once.

I think it would be helpful to have a set-like ADT in strongSwan or enhance `array.c` with a `contains` method. This would reduce my nested loops. I am absolutely open to other approaches though.